compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNTypes.cpp

```diff
@@ -45,3 +53,96 @@ void CUDNNDialect::registerTypes() {
 #include "openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNTypes.cpp.inc"
       >();
 }
+
+//===----------------------------------------------------------------------===//
+// !cudnn.tensor
+//===----------------------------------------------------------------------===//
+
+namespace openxla::compiler::nvgpu::cudnn {
+
+bool TensorType::isOpaque() {
+  return getShape().empty() && !getElementType() && !getLayout() &&
+         !getStrides();
+}
+
+LogicalResult TensorType::verify(function_ref<InFlightDiagnostic()> emitError,
```

Comment:
Would it make sense to cover these errors by tests?

Reply:
It triggers assertion error and crashes. I'll try to add them later once I'll find how to do it without crashing the tool.